### PR TITLE
[console-unix] retry `tcsetattr` if it was interrupted by signal

### PR DESCRIPTION
**Pull request by lewurm**
if configuring the terminal failed, we still try to continue (see
a3ce2f008bdb2fa5108034f1687fc7f87c92a0b8 ) and therefore succedings
`read`s can fail.

Fixes https://bugzilla.xamarin.com/show_bug.cgi?id=26983

### CODE REVIEW

**Comment by lewurm** [comment]
@coop guys: does the syscall need a `MONO_ENTER_GC_SAFE`/`MONO_EXIT_GC_SAFE` guard as other syscalls in this file?

**Comment by vargaz** [comment]
Probly yes, if it can block.

File: mono/metadata/console-unix.c [hunk 1]
```diff
@@ -463,7 +463,14 @@ ves_icall_System_ConsoleDriver_TtySetup (MonoStringHandle keypad, MonoStringHand
...
-	if (tcsetattr (STDIN_FILENO, TCSANOW, &mono_attr) == -1)
+	gint ret;
+	do {
+		MONO_ENTER_GC_SAFE;
+		ret = tcsetattr (STDIN_FILENO, TCSANOW, &mono_attr);
+		MONO_EXIT_GC_SAFE;
+	} while (ret == -1 && errno == EINTR);
+
+	if (ret == -1)
 		return FALSE;
```
Comment (migueldeicaza) [inline mono/metadata/console-unix.c]: Whoa!   very good catch!   This is indeed broken code.

**Comment by lewurm** [comment]
@monojenkins squash

**Pull request merged**